.github/workflows/schedule.yml

```diff
@@ -0,0 +1,281 @@
+on:
+  # Daily trigger to check updates
+  schedule:
+    - cron: "0 0 * * *"
+  workflow_dispatch:
+    inputs:
+      dry-run:
+        description: "Run tests but do not create issues {true,false}"
+        required: true
+        default: "true"
+      vendor-type:
+        description: "Vendor type {all,partner,redhat,community}"
+        required: true
+        default: "all"
+      notify-id:
+        description: "(Optional) Issue notification {github id}"
+        required: false
+        default: ""
+jobs:
+  check-ocp:
+    name: Check OpenShift Version
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Install oc
+        run: |
+          curl -sLO https://mirror.openshift.com/pub/openshift-v4/clients/ocp/stable/openshift-client-linux.tar.gz
+          tar zxvf openshift-client-linux.tar.gz oc
+
+      - name: Log into OpenShift cluster
+        run: |
+          API_SERVER=$(echo -n ${{ secrets.API_SERVER }} | base64 -d)
+          ./oc login --insecure-skip-tls-verify --token=${{ secrets.CLUSTER_TOKEN }} --server=${API_SERVER}
+        shell: bash
+
+      - name: Get current OpenShift version
+        id: get_curr_ocp_version
+        run: |
+          OCP_VERSION=$(./oc version -o json | jq '.releaseClientVersion')
+          printf "[INFO] Current OCP Version: %s\n" ${OCP_VERSION}
+          echo "::set-output name=curr_ocp_version::${OCP_VERSION}"
+        shell: bash
+
+      - name: Checkout software-version branch
+        uses: actions/checkout@v2
+        with:
+          ref: "software-version"
+
+      - name: Read previous OpenShift version
+        id: get_prev_ocp_version
+        uses: mikefarah/yq@master
+        with:
+          cmd: yq e '.openshift.release-client-version' software-version.yaml
+
+      - name: Compare OpenShift versions
+        id: compare_ocp_versions
+        run: |
+          set -euo pipefail
+          if [ "${{ steps.get_curr_ocp_version.outputs.curr_ocp_version }}" = "${{ steps.get_prev_ocp_version.outputs.result }}" ]; then
+            # No change in the OpenShift version
+            printf "OpenShift version has not changed since last run: '%s' -> '%s'\n" "${{ steps.get_prev_ocp_version.outputs.result }}" "${{ steps.get_curr_ocp_version.outputs.curr_ocp_version }}"
+            echo "::set-output name=run_tests::false"
+          else
+            # New OpenShift version is set
+            printf "OpenShift version has changed since last run: '%s' -> '%s'\n" "${{ steps.get_prev_ocp_version.outputs.result }}" "${{ steps.get_curr_ocp_version.outputs.curr_ocp_version }}"
+            echo "::set-output name=run_tests::true"
+          fi
+        shell: bash
+
+      - name: Update software-version.yaml
+        if: |
+          steps.compare_ocp_versions.outputs.run_tests == 'true'
+        uses: mikefarah/yq@master
+        with:
+          cmd: yq eval -i '.openshift.release-client-version = ${{ steps.get_curr_ocp_version.outputs.curr_ocp_version }}' 'software-version.yaml'
```

Comment:
This will create the file if it is not there right?

Reply:
No it will throw error when the file does not exist (which it should exist).

Reply:
See question above